src/CommonUtils.jl

```diff
@@ -2,10 +2,11 @@
 
 
 function resid2DLinear(μ, mus, Lambdas; diffop::Function=-)  # '-' exploits EuclideanManifold commutativity a-b = b-a
-  dμ = broadcast(diffop, μ, mus)  # mus .- μ  ## μ .\ mus
+  # dμ = broadcast(diffop, μ, mus)  # mus .- μ  ## μ .\ mus
   # @show round.(dμ, digits=4)
-  ret = sum( Lambdas.*dμ )
-  return ret
+  # ret = sum( Lambdas.*dμ )
+  r = map((mu, lam) -> diffop(μ[], mu) * lam, mus, Lambdas)
```

Comment:
@dehann, here is a version without the extra dispatch and allocations. 
Is μ always of length 1?

Reply:
I think so, will double check bit later today.

Reply:
looks right